src/language/class-modifiers.md

```diff
@@ -0,0 +1,240 @@
+Only the `base` modifier can appear before a mixin declaration. The modifiers do
+not apply to other declarations like `enum`, `typedef`, or `extension`.
+
+When deciding whether or not to use class modifiers, consider the following questions:
+1. Is [any access](#no-modifier) fine?
+2. If not, what do you want to restrict?
+    - [Implementation](#base)?
+    - [Inheritance](#interface)?
+    - [Both](#final)?
+    - [Subtyping](#sealed) by outside libraries at all?
+3. Then, does it need to be [abstract](#abstract)?
+4. And lastly, should it be a [class, mixin, or mixin class](#class-mixin-or-mixin-class)?
+
+## No modifier
+
+Use a `class` or `mixin` declaration without a modifier to **allow unrestricted
+access** to outside libraries. The standard behaviors apply:
+
+- [Construct][] new instances of a class
+- [Extend][] a class to create a new subtype
+- [Implement][] a class's interface
+- [Mix in][] a mixin or mixin class
+
+## `base`
+
+Use the `base` modifier to **disallow implementation** of a class or mixin.
+A base class can be extended and constructed. This applies trasitively to all
+subtypes, because implementing a subtype implements the supertype as well.  
+
+{{site.alert.note}}
+  Only the `base` modifier can accompany a mixin declaration. 
+  - A `base mixin` can be mixed in, but not implemented.
+  - A `base mixin class` can be constructed, extended, and mixed in, but not
+  implemented.
+{{site.alert.end}}
+
+```dart
+base class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+var myCar = Vehicle(     // Can be constructed
+  make: 'Ford',
+  model: 'T',
+);
+
+class Car extends     // Can be extended
+  Vehicle {
+    int passengers;
+…
+}
+
+class MockVehicle     // ERROR: Cannot be implemented
+  implements Vehicle {
+    @override
+    void moveForward …
+}
+
+```
+
+Disallowing implementation makes `base` useful for:
+- Preventing users of the API from overriding the constructor, for example, to
+bypass its validation.
+- Ensuring all private members are available to subtypes of the base class, and
+and accessing them will never cause a [`noSuchMethod`][] error.
+
+## `interface`
+
+Use the `interface` modifier to **disallow extending a class**.
+An interface class can be implemented and constructed.
+
+```dart
+interface class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+var myCar = Vehicle(     // Can be constructed
+  make: 'Ford',
+  model: 'T',
+);
+
+class Car extends     // ERROR: Cannot be inherited
+  Vehicle {
+    int passengers;
+…
+}
+
+class MockVehicle     // Can be implemented     
+  implements Vehicle {
+    @override
+    void moveForward …
+}
+```
+
+Disallowing inheritance makes `interface` useful for:
+- Defining API interfaces.
+- Preventing users of the API from overriding a subset of methods.
+
+## `final` 
+
+Use the `final` modifier to **disallow extending _and_ implementing** a class.
```

Comment:
Might be worth saying that the transitive base property applies here as well?  That is, something like "within the same library, a final class can be extended or implemented, but the subtype is required to be marked base".  Or are you avoiding talking about the "same library" exceptions entirely?

Reply:
I unintentionally avoided talking about the same library, because I was unclear on whether that functionality was something worth talking about, but now I've got it! I'll add this, thanks